azure-pipelines.yml

```diff
@@ -0,0 +1,83 @@
+stages:
+ - stage: check
+   displayName: Compilation check
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+         name: cargo_check
+         benches: true
+ # This represents the minimum Rust version supported.
+ # Tests are not run as tests may require newer versions of rust.
+ - stage: msrv
+   displayName: "Minimum supported Rust version"
+   dependsOn: []
+   jobs:
+     - template: azure/cargo-check.yml@templates
+       parameters:
+        rust: 1.34.0
+ - stage: custom
+   displayName: "Special tests"
+   dependsOn: check
+   jobs:
+     - job: custom_stable
+       pool:
+         vmImage: ubuntu-16.04
+       displayName: "Stable custom tests"
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: stable
+         - bash: cd tracing/test-log-support && cargo test
+           displayName: "Test log support"
+         - bash: cd tracing/test_static_max_level_features && cargo test
+           displayName: "Test static max level features"
+     - job: custom_nightly
+       pool:
+         vmImage: ubuntu-16.04
+       displayName: "Nightly custom tests"
+       continueOnError: true
+       steps:
+         - template: azure/install-rust.yml@templates
+           parameters:
+            rust: nightly
+         - script: cargo test --features=doctest-readme --all
+           displayName: "Test README as doctest"
+         - bash: cd tracing-futures/test_std_future && cargo test
+           displayName: "Test std::future support for tracing-futures"
+ - stage: style
+   displayName: Style linting
+   dependsOn: check
+   jobs:
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt
+    - template: azure/rustfmt.yml@templates
+      parameters:
+        name: rustfmt_beta
+        rust: beta
+        allow_fail: true
+    - template: azure/cargo-check.yml@templates
+      parameters:
+        name: warning_free
+        err_on_warn: true
+        allow_fail: true
+ - stage: test
+   displayName: Test suite
+   dependsOn: check
+   jobs:
+     - template: azure/tests.yml@templates
+ - stage: coverage
+   displayName: Code coverage
+   dependsOn: test
+   jobs:
+     - template: azure/coverage.yml@templates
+       parameters:
+         codecov_token: $(CODECOV_TOKEN_SECRET)
+
+resources:
+  repositories:
+    - repository: templates
+      type: github
+      name: crate-ci/azure-pipelines
+      endpoint: tokio-rs
```

Comment:
@hawkw @carllerche to make this work I need to know the name of your GitHub service connection in Azure Pipelines. See the second paragraph [here](https://github.com/crate-ci/azure-pipelines#each-time-setup).

Reply:
@jonhoo https://dev.azure.com/tracing/tracing/

Reply:
@hawkw ah, no, I need the service connection name:
> Go to "Project settings" (bottom left), the "Service connections" (under "Pipelines"). There should be one thing listed there, and it's your authenticated connection to GitHub. Note down its name.